datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -299,11 +302,12 @@ public void start() {
   public void stop() {
     _log.info("Stopping coordinator");
 
+    _shutdown = true;
     // Stopping event threads so that no more events are scheduled for the connector.
     while (_eventThread.isAlive()) {
       try {
+        _eventThread.join(EVENT_THREAD_JOIN_TIMEOUT.toMillis());
```

Comment:
I think we should move this to line 306; we first wait for the thread to finish (join) then we start interrupting it if it still alive.

Reply:
Changed according to the feedbacks, I don't think we need a timeout for the second join as this will cause leaking. I will just add the first section.